File: ios/ShortDrama/Sources/Data/DTOs/HotSearchDTO.swift
import Foundation

/// DTO for a hot search item.
struct HotSearchItemDTO: Codable, Equatable {
    let rank: Int
    let keyword: String
    let score: Int
}

extension HotSearchItemDTO {
    func toEntity() -> HotSearchItem {
        HotSearchItem(rank: rank, keyword: keyword, score: score)
    }
}

/// API response for the hot search endpoint.
struct HotSearchListResponseDTO: Codable, Equatable {
    let data: [HotSearchItemDTO]
}

File: android/app/src/main/java/com/djs66256/short_drama/data/dto/HotSearchDto.kt
package com.djs66256.short_drama.data.dto

import com.djs66256.short_drama.domain.model.HotSearchItem
import kotlinx.serialization.Serializable

@Serializable
data class HotSearchItemDto(
    val rank: Int,
    val keyword: String,
    val score: Int,
) {
    fun toDomain(): HotSearchItem = HotSearchItem(
        rank = rank,
        keyword = keyword,
        score = score,
    )
}

@Serializable
data class HotSearchListResponseDto(
    val data: List<HotSearchItemDto>,
)
